WDB-45-114-112-127-140/Finish recycling bin page and improve website responsiveness

kibartas opened the pull request:
Dadam. Probably last PR in this beautiful project. Edit: Forgot that we didn't do logging yet

So, I had a problem with the Recycling/Library pages. They were way too similar to not do anything and leave them both independent as is. 

Although I don't think my answer is perfect (feedback appreciated) I decided to create a third component which would be pure and accept A LOOOOOOOOOOOOOOT of props from LibraryPage and RecyclingBinPage. See for yourselves how that's implemented

Also, finished the API. Added the restore functionality and the delete forever functionality in Front.

Renamed a few things here and there. 

Endpoints "restore" and "delete" now return ID's in the response. This is so the pages could be more responsive and wouldn't need to do another request just to get which cards were actually deleted or restored.

Commit:
[WDB-112]
change \ to Path.Combine

Commit:
[WDB-112]
change applicationUrl

Commit:
[WDB-112]
move VideoCardPage utility functions to utils

Commit:
[WDB-112]
remove babel-eslint parser and dependency

Commit:
[WDB-112]
restructure LibraryPage and RecyclingBinPage, add API calls, execute them in the beforementioned containers

Commit:
[WDB-112]
change VideoCardPage from class to function

AArlauskas commented on backend/BusinessLogic/Services/VideoService/VideoService.cs:
```diff
@@ -99,7 +99,7 @@ public async Task<Video> CompleteUpload(Guid userId, string fileName)
         public async Task DeleteVideo(Video video, Guid userId)
         {
-            string snapshotPath = Path.Combine(Path.Combine(_uploadPath, userId.ToString()), "Snapshots\\" + video.Id + ".png");
+            string snapshotPath = Path.Combine(Path.Combine(_uploadPath, userId.ToString()), Path.Combine("Snapshots", video.Id + ".png"));
```
How was this still here??

kibartas replied:
No fucking idea

AArlauskas commented on frontend/src/components/VideoCardPage/styles.css:
```diff
@@ -0,0 +1,12 @@
+    margin: auto !important;
+}
+
+
```
EMPTY LINE @tmaciulis22 PLEASE RESPOND WHAT SHOULD WE DOOOO

tmaciulis22 replied:
SHAME!!!!

AArlauskas commented on backend/BusinessLogic/Services/VideoService/VideoService.cs:
```diff
@@ -99,7 +99,7 @@ public async Task<Video> CompleteUpload(Guid userId, string fileName)
 
         public async Task DeleteVideo(Video video, Guid userId)
         {
-            string snapshotPath = Path.Combine(Path.Combine(_uploadPath, userId.ToString()), "Snapshots\\" + video.Id + ".png");
```
General Comment: 
![image](https://user-images.githubusercontent.com/31959982/119029916-eb1f9e80-b9b1-11eb-8b4e-5d4dafd8a792.png)

It happens after restoring the last deleted video

kibartas replied:
what the fuck

AArlauskas commented on backend/BusinessLogic/Services/VideoService/VideoService.cs:
```diff
@@ -99,7 +99,7 @@ public async Task<Video> CompleteUpload(Guid userId, string fileName)
-            string snapshotPath = Path.Combine(Path.Combine(_uploadPath, userId.ToString()), "Snapshots\\" + video.Id + ".png");
+            string snapshotPath = Path.Combine(Path.Combine(_uploadPath, userId.ToString()), Path.Combine("Snapshots", video.Id + ".png"));
 
             if (!File.Exists(video.Path))
```
General comment: 

https://user-images.githubusercontent.com/31959982/119030190-3c2f9280-b9b2-11eb-831b-efddaa30271d.mp4

Some flickering action going on

AArlauskas replied:
You should wait for a response before rendering anything

AArlauskas commented on frontend/src/components/VideoCardPage/VideoCardPage.js:
```diff
@@ -0,0 +1,122 @@
+  selectedCardIds,
+  showDeletionError,
+  iconsToShow,
+  iconsToShowOnSelected,
```
Not sure I like that many props being passed down, I think it's a bit overkill. But you already did it, so I don't think you should be reversing it back

AArlauskas reviewed (changes requested):
Good job, but still needs some fixing

AArlauskas commented on frontend/yarn.lock:
```diff
@@ -5176,6 +5176,11 @@ eslint-visitor-keys@^2.0.0:
   resolved "https://registry.yarnpkg.com/eslint-visitor-keys/-/eslint-visitor-keys-2.0.0.tgz#21fdc8fbcd9c795cc0321f0563702095751511a8"
   integrity sha512-QudtT6av5WXels9WjIM7qz1XD1cWGvX4gGXvp/zBn9nXG02D0utdU3Em2m/QjTnrsk6bBjmCygl3rmj118msQQ==
```
Nothing

AArlauskas replied:
to

AArlauskas replied:
fix

AArlauskas replied:
here

AArlauskas replied:
.

AArlauskas replied:
Just

AArlauskas replied:
adding

AArlauskas replied:
more

AArlauskas replied:
comments

AArlauskas replied:
to

AArlauskas replied:
make

AArlauskas replied:
it

AArlauskas replied:
look

AArlauskas replied:
worse

AArlauskas replied:
and

AArlauskas replied:
to

AArlauskas replied:
get

AArlauskas replied:
back

AArlauskas replied:
at

AArlauskas replied:
you

AArlauskas replied:
for

AArlauskas replied:
making

AArlauskas replied:
so

AArlauskas replied:
many

AArlauskas replied:
comments

AArlauskas replied:
in

AArlauskas replied:
my

AArlauskas replied:
Pull

AArlauskas replied:
Requests

AArlauskas replied:
.

tmaciulis22 replied:
❤️ ❤️ ❤️

tmaciulis22 replied:
Let's

tmaciulis22 replied:
Make

tmaciulis22 replied:
a

tmaciulis22 replied:
new

tmaciulis22 replied:
record

tmaciulis22 replied:
of

tmaciulis22 replied:
PR

tmaciulis22 replied:
Comments

tmaciulis22 commented on frontend/src/components/VideoCardPage/VideoCardPage.js:
```diff
@@ -0,0 +1,122 @@
+import DeleteConfirmationDialog from '../DeleteConfirmationDialog/DeleteConfirmationDialog';
+
+const VideoCardPage = ({
+  renderSnackbars,
+  title,
+  size,
+  showNavDrawer,
+  videosInformation,
+  showUploadModal,
+  showDeletionDialog,
+  selectedCardIds,
+  showDeletionError,
+  iconsToShow,
+  iconsToShowOnSelected,
```
Wth is this monstrosity??

kibartas replied:
Yeah, so, I didn't know how to accomplish not repeating the same code in two places. Any suggestions how to do that, for the future? I mean, how would you think you woulda done it here. Just leave the two extremely similar containers as-is, or pull something out

kibartas replied:
Added spinner

tmaciulis22 reviewed (changes requested):
Speaking from professional experience, these types of UI components used accross multiple use cases, often times just end up being useless, because two different usecases most of the time at some point in app lifecycle end up being so different that it is impossible to implement them with one big ass component. But I guess since you've done it, then so be it. But fix the bugs found by @AArlauskas

tmaciulis22 commented on frontend/src/components/VideoCardPage/VideoCardPage.js:
```diff
@@ -0,0 +1,122 @@
+        <CustomSnackbar
+          message={
+            selectedCardIds.length === 1
+              ? 'Oops... Something wrong happened. We could not delete your video'
+              : 'Oops... Something wrong happened. Some videos might not have been deleted'
```
Wait, so we reuse same messages for deletion and permanent deletion?

kibartas replied:
Changed

AArlauskas commented on frontend/src/components/VideoCardPage/styles.css:
```diff
@@ -0,0 +1,12 @@
+}
+
+.card_container {
+    margin: auto !important;
```
You know that !important is a bad practice. No way around it?

kibartas replied:
Not that I found

Commit:
[WDB-112]
add loading spinner

Commit:
[WDB-112]
PR fixes

tmaciulis22 reviewed (changes requested):
✅**EDIT: FIXED**
  Add spinner to register page, since usually you have to wait quite a while until email service sends an email, so user is left hanging there on that page.

tmaciulis22 reviewed (commented):
✅**EDIT: FIXED**
Some unused vars  
Deletion error snackbar is missing in `LibraryPage`

tmaciulis22 reviewed (commented):
✅**EDIT: FIXED**

Also, regarding the deletedate in backend. So we are still using it to show when will the video will be deleted, or when it was moved to recycling bin?

tmaciulis22 reviewed (commented):
✅**EDIT:** This bug has been fixed
Found a bug with login. If I try to login with a user which doesn't exist, the server error is thrown and page is reloaded. I think it is because of the interceptor, which you have implemented previously.

tmaciulis22 committed:
[WDB-112] Fix mobile layout issues in TopBar

tmaciulis22 reviewed (commented):
✅**EDIT:** This issue has been fixed  

Not sure whether this stuff happens on my device, but if I move every video to recycling bin, then empty library page's layout is incorrect 
![image](https://user-images.githubusercontent.com/29150754/119151319-f1685600-ba57-11eb-8d57-d84e2df2d7f8.png)

tmaciulis22 committed:
[WDB-112] Fix empty VideoCardPage content layout issues

tmaciulis22 commented on frontend/src/components/VideoCardPage/VideoCardPage.js:
```diff
@@ -0,0 +1,114 @@
+import React from 'react';
```
⚠⚠⚠Found another bug. If I delete all the videos from recycling bin, go to user profile page and click the arrow icon back in top bar, then all the deleted videos reappear again. It has possibly to do something with react router history prop and it's goBack function

tmaciulis22 committed:
[WDB-112] Remove spacing from some TopBar items

tmaciulis22 commented on backend/RestAPI/Controllers/VideosController.cs:
```diff
@@ -393,6 +393,8 @@ public async Task<ActionResult> MarkVideosForDeletion([FromBody] List<Guid> ids)
                 return NotFound();
             }
 
+            var markedVideos = new LinkedList<Guid>();
```
⚠⚠⚠Another bug. If you go to recycling bin, click on deleted video card, it opens video player page. It should not open this page.

tmaciulis22 committed:
[WDB-112] Fix some bugs caused by interceptor

tmaciulis22 commented on frontend/src/api/Api.js:
```diff
@@ -13,6 +13,7 @@ Api.interceptors.response.use(
       sessionStorage.clear();
       window.location.href = '/';
     }
+    return Promise.reject(error);
```
I added this line because at some cases, when client receives error response from backend, then it is not delivered from this interceptor to the place where request was made. So, all the catch blocks did not work. This line fixes the problem.

tmaciulis22 commented on backend/RestAPI/Controllers/UsersController.cs:
```diff
@@ -109,7 +109,7 @@ public async Task<ActionResult<string>> Authenticate(string email, string passwo
             if (token is null)
             {
-                return Unauthorized();
+                return NotFound();
```
I changed this on purpose. Since on this method we check whether user exists or not, so I think it makes sense that it returns `NotFound`. Also, this way the response won't get picked by interceptor in front-end.

tmaciulis22 commented on frontend/src/components/TopBar/TopBar.js:
```diff
@@ -46,11 +46,7 @@ export default function TopBar({
-        {iconItems}
-      </Grid>
-    );
+    return iconItems;
```
I removed some spacing of icons, so all of them could fit on mobile devices

tmaciulis22 replied:
Same applies to the changes in this component below

tmaciulis22 commented on frontend/src/containers/LoginPage/LoginPage.js:
```diff
@@ -40,7 +40,7 @@ class LoginPage extends React.Component {
         }
         const { status } = ex.response;
-        if (status === 401) this.setState({ showWrongCredentialsError: true });
+        if (status === 404) this.setState({ showWrongCredentialsError: true });
```
This was changed due to backend change I did on `authenticate` endpoint.

tmaciulis22 commented on frontend/src/containers/LibraryPage/LibraryPage.js:
```diff
@@ -243,101 +244,74 @@ class LibraryPage extends React.Component {
+        );
+      }
+
+      if (showDeletionSuccess) {
```
⚠⚠⚠ Deletion(move to recycling bin) error snackbar is missing

Commit:
[WDB-112]
add spinner to registerpage

Commit:
[WDB-112]
fix goBack issue with no more caching

Commit:
[WDB-112]
change recycling bin video player to not include download and mark for deletion buttons

kibartas committed:
Merge branch 'master' into WDB-112/Create-recycling-bin-with-videos

Commit:
[WDB-112]
rollback appsettings.json

Commit:
Merge branch 'WDB-112/Create-recycling-bin-with-videos' of github.com:kibartas/PSK-Front-End into WDB-112/Create-recycling-bin-with-videos

tmaciulis22 committed:
[WDB-112] Add log.txt file to gitignore

tmaciulis22 commented on backend/RestAPI/Program.cs:
```diff
@@ -1,3 +1,4 @@
+using System;
```
Why this was added?

kibartas replied:
remove unused imports

tmaciulis22 commented on backend/RestAPI/Properties/launchSettings.json:
```diff
@@ -22,7 +22,7 @@
       "launchBrowser": true,
       "launchUrl": "swagger",
-      "applicationUrl": "https://localhost:5001;http://localhost:5000",
+      "applicationUrl": "https://localhost:5001;http://localhost:61346",
```
Why this was changed?

tmaciulis22 commented on backend/RestAPI/Startup.cs:
```diff
@@ -104,8 +104,8 @@ public void ConfigureServices(IServiceCollection services)
                 app.UseSwaggerUI(c => c.SwaggerEndpoint("/swagger/v1/swagger.json", "backend v1"));
             }
 
+            app.UseHttpsRedirection();
```
Why this was added?

tmaciulis22 commented on frontend/src/api/Api.js:
```diff
@@ -1,9 +1,18 @@
 import axios from 'axios';
 
-const baseUri = 'http://localhost:61346/api';
+let baseUri;
+if (!process.env.NODE_ENV || process.env.NODE_ENV === 'development') {
+  baseUri = 'http://localhost:61346/api';
+} else {
+  baseUri = 'http://13.81.9.144/api';
+}
 
 const Api = axios.create({
   baseURL: baseUri,
+  headers: {
+    'Cache-control': 'no-cache',
+    Pragma: 'no-cache',
+  },
```
Why this was added/changed?

kibartas replied:
When history.goBack() is done the browser returns cached request to the frontend. That's why you saw videos that you had already deleted in Recycling Bin when you went to the profile page and back. 

This here tells the browser to not cache any requests

tmaciulis22 committed:
[WDB-112] Delete prod stuff

tmaciulis22 commented on backend/RestAPI/Controllers/VideosController.cs:
```diff
@@ -313,14 +313,16 @@ public async Task<IActionResult> UploadComplete(string fileName)
-                catch
+                catch (Exception ex)
                 {
+                    Console.WriteLine(ex);
```
Why this was added?

tmaciulis22 commented on backend/RestAPI/Controllers/VideosController.cs:
```diff
@@ -313,14 +313,16 @@ public async Task<IActionResult> UploadComplete(string fileName)
             {
                 try
                 {
+                    Console.WriteLine(e);
```
Why this was added?

tmaciulis22 commented on frontend/src/components/VideoCardPage/VideoCardPage.js:
```diff
@@ -0,0 +1,115 @@
+        onClose={toggleNavDrawer}
+        spaceTaken={size}
+      />
+      <UploadModal show={showUploadModal} onClose={toggleUploadModal} />
```
Move this to Library page component, since it won't be used in RecyclingBinPage. Also, I found a bug when I upload two videos and change one video title, then the other one disappears from modal, since modal wasn't changed, I think that this bug might be fixed if you would move the UploadModal to LibraryPage

kibartas replied:
Don

tmaciulis22 commented on frontend/src/containers/RecyclingBinPage/RecyclingBinPage.js:
```diff
@@ -1,57 +1,304 @@
+        selectedCardIds={selectedCardIds}
+        iconsToShow={[SelectAllIcon, SortIcon]}
+        handleIconsClick={[this.toggleSelectAll, this.toggleSort]}
+        iconsToShowOnSelected={[RestoreIcon, DeleteIcon]}
```
Very minor thing, but different delete icon is used than one in zeplin

tmaciulis22 replied:
And same in VideoPlayer if you open in it from RecyclingBinPage

tmaciulis22 commented on frontend/src/containers/RecyclingBinPage/RecyclingBinPage.js:
```diff
@@ -1,57 +1,304 @@
+    this.setState({ showDeletionDialog: !showDeletionDialog });
+  };
+
+  handleVideoDeletion = () => {
```
Bug. When I delete forever a video, then total space taken is not updated. It still includes the forever deleted video's size

kibartas replied:
Fixed

tmaciulis22 committed:
[WDB-112] Fix memory leak in RegisterPage and event handler naming in PlayerPage

Commit:
[WDB-112]
remove unused imports

Commit:
[WDB-112]
fix bug with deleteDate in recycling bin, get new size after cards get deleted

Commit:
Merge branch 'WDB-112/Create-recycling-bin-with-videos' of github.com:kibartas/PSK-Front-End into WDB-112/Create-recycling-bin-with-videos

Commit:
[WDB-112]
take out uploadmodal out of VideoCardPage and put it in LibraryPage

tmaciulis22 commented on frontend/src/containers/RegisterPage/RegisterPage.js:
```diff
@@ -23,22 +26,26 @@ class RegisterPage extends React.Component {
         const { status } = ex.response;
         if (status === 409) this.setState({ showUserExistsError: true });
         else this.setState({ showGeneralError: true });
+      })
+      .finally(() => {
       });
```
Empty finally block

tmaciulis22 replied:
It was left by me I think, but could you please delete it?

kibartas replied:
CXDDDDDDDDDDDd

tmaciulis22 commented on backend/RestAPI/Controllers/VideosController.cs:
```diff
@@ -313,13 +313,13 @@ public async Task<IActionResult> UploadComplete(string fileName)
                 return Ok(response);
             }
-            catch(Exception)
+            catch(Exception e)
```
Is this exception used somehwere?

kibartas replied:
XDDDDDDDDDDDD

tmaciulis22 commented on backend/RestAPI/Controllers/VideosController.cs:
```diff
@@ -313,13 +313,13 @@ public async Task<IActionResult> UploadComplete(string fileName)
                     _videoService.DeleteAllChunks(fileName);
                 }
-                catch
+                catch (Exception ex)
```
Is this exception used somehwere?

kibartas replied:
Xd

tmaciulis22 reviewed (changes requested):
Minor stuff to look at, but I think the moment of merging this PR is getting closer and closer 👍🏻

tmaciulis22 commented on frontend/src/containers/PlayerPage/PlayerPage.js:
```diff
@@ -218,20 +257,39 @@ class PlayerPage extends React.Component {
+                  !isFromBin
+                    ? [
+                        this.toggleInformationDrawer,
+                        handleVideoDownload,
```
⚠⚠⚠For some reason video download does not work anymore

kibartas committed:
[WDB-112] remove unnecessary exception clauses from videoController

kibartas committed:
delete Tautvis' finally block

Commit:
[WDB-112]
merge master

Commit:
[WDB-112]
change sessionStorage to localStorage in VideoCardPage (part of merge, I think)

Commit:
Merge branch 'WDB-112/Create-recycling-bin-with-videos' of github.com:kibartas/PSK-Front-End into WDB-112/Create-recycling-bin-with-videos

tmaciulis22 commented on backend/RestAPI/appsettings.json:
```diff
@@ -1,6 +1,6 @@
 {
   "ConnectionStrings": {
-    "BackendContext": "Server=.\\SQLEXPRESS;Database=WeDontByte;Trusted_Connection=True;"
+    "BackendContext": "Server=tcp:wedontbyte.database.windows.net,1433;Initial Catalog=WeDontByte;Persist Security Info=False;User ID=kibartas;Password=^zPXamfe9Tgr2g2lbV&3DvEMNNXSD;MultipleActiveResultSets=False;Encrypt=True;TrustServerCertificate=False;Connection Timeout=30;"
```
I think this goes to prod

tmaciulis22 committed:
[WDB-112] Fix download error in PlayerPage

tmaciulis22 commented on frontend/src/containers/RecyclingBinPage/RecyclingBinPage.js:
```diff
@@ -1,57 +1,311 @@
+    this.setState({ showDeletionDialog: !showDeletionDialog });
+  };
+
+  handleVideoDeletion = () => {
```
There is a bug in this page and in `LibraryPage`. If you delete a video, then go to `ProfilePage` and go back to previous page, then the deleted video keeps appearing again

kibartas replied:
Did you solve this?

tmaciulis22 replied:
No

tmaciulis22 reviewed (changes requested):
I think these comments will be the last ones from me 🤞🏻

tmaciulis22 committed:
[WDB-112] Change reset password car button text

tmaciulis22 committed:
[WDB-112] Change namings in ResetPasswordPage

AArlauskas commented on backend/RestAPI/Startup.cs:
```diff
@@ -105,7 +105,6 @@ public void ConfigureServices(IServiceCollection services)
             }
 
             app.UseMiddleware<LoggingMiddleware>();
-            //app.UseHttpsRedirection();
```
General comment: I think we should keep this one in production. Right?

kibartas replied:
lol, this is not the production branch. But yes

AArlauskas commented on frontend/src/api/Api.js:
```diff
@@ -4,6 +4,10 @@ const baseUri = 'http://localhost:61346/api';
 
 const Api = axios.create({
   baseURL: baseUri,
+  headers: {
```
Could you explain this one please?

kibartas replied:
Ask the browser not to cache the results of requests. If the user goes back to the page he was before, the browser pulls cached responses. With this, it doesn't. At least on Firefox, @tmaciulis22 told me that it doesn't work on Chrome

AArlauskas commented on frontend/src/components/TopBar/TopBar.js:
```diff
@@ -61,7 +57,7 @@ export default function TopBar({
       >
         <Toolbar>
-          <Grid container alignItems="center" spacing={1}>
+          <Grid container direction="row" alignItems="center">
```
direction="row" is not needed as it is the default value of Grid component

kibartas replied:
I like the explicit way

AArlauskas commented on frontend/src/containers/RegisterPage/RegisterPage.js:
```diff
@@ -73,7 +78,11 @@ class RegisterPage extends React.Component {
           <Grid item xs={10} sm={6} md={4}>
-            <RegistrationForm onRegister={this.handleRegister} />
+            {isLoading ? (
+              <LoadingSpinner />
```
Thanks. A loading spinner was really needed

kibartas committed:
Revert appsettings.json change

kibartas committed:
hotfix in DB string

Commit:
[WDB-112]
bugfix: profile page goback button now returns you to lib or bin correctly (ghost videoCards are no longer present). Change some window.location.href to history.push() for performance reasons

Commit:
Merge branch 'WDB-112/Create-recycling-bin-with-videos' of github.com:kibartas/PSK-Front-End into WDB-112/Create-recycling-bin-with-videos

Commit:
[WDB-112]
make the project a lot more smooth. Remove href and add Links and history.push()

tmaciulis22 committed:
[WDB-112] Make upload modal reactive

tmaciulis22 reviewed (changes requested):
wrong choice. Meant to click approve, oopsie

tmaciulis22 reviewed (approved):
Finally it is good to merge 👏🏻👏🏻👏🏻

tmaciulis22 committed:
[WDB-112] Fix eslint rule error
